### PR TITLE
Feat/finding info in appmap

### LINKED ISSUE #876: Finding details are added to map view in IDE
Analysis finding information is added into the map-view in the IDE.

## Findings
Image

## Finding details
Image

### PR DESCRIPTION
**Pull request by ahtrotta**
Fixes #876 
Replaces #868

Now findings information can be viewed within an AppMap (if it has one or more associated findings).

I haven't written tests for this change yet, but I'm planning on doing that as my first task in the new year.

Each finding appears in the details panel:

image

When clicked, the trace filter gets populated and the user can iterate through the events associated with the clicked finding:

image

Each event also displays more detailed information about the finding in the details panel:

image

### CODE REVIEW

File: packages/components/src/components/DetailsSearch.vue [hunk 1]
```diff
@@ -264,57 +319,81 @@ export default {
...
     }
 
     &-list {
       margin: 0;
-      padding: 0;
+      padding: 0 0.5rem;
       list-style: none;
+      li {
+        &:last-of-type {
+          border-bottom: 0;
```
Comment (dustinbyrne) [inline packages/components/src/components/DetailsSearch.vue:374]: This makes the last label look like it's being clipped
image

File: packages/components/src/components/DetailsSearch.vue [hunk 2]
```diff
@@ -179,6 +210,13 @@ export default {
     selectObject(type, object) {
       if (type === 'labels') {
         this.$store.commit(SELECT_LABEL, object);
+      } else if (type === 'analysis-finding') {
+        const { events } = object;
+        if (events && events.length >= 1) {
+          this.$store.commit(SELECT_OBJECT, events);
```
Comment (dustinbyrne) [inline packages/components/src/components/DetailsSearch.vue:216]: I'd expect to be brought to finding details, not an event. A finding may consist of many events and the finding details panel should help us navigate them.

File: packages/components/src/components/FindingDetailsPanel.vue [hunk 3]
```diff
@@ -0,0 +1,274 @@
...
+    padding: 0 1rem;
+    .code {
+      color: $white;
+      margin-top: 0;
+      font-family: monospace;
+    }
+  }
+  .heading,
+  section {
+    border-bottom: 3px solid $gray2;
```
Comment (dustinbyrne) [inline packages/components/src/components/FindingDetailsPanel.vue:144]: There's some inconsistency here with the border width. The outer border is `1px` while the inner is `3px`. We're also not adding an outer border anywhere else, but I think this is more of an issue of the finding details needing to be its own panel.
image

File: packages/components/src/components/DetailsSearch.vue (hunk 1 above)
Comment (dustinbyrne) [reply packages/components/src/components/DetailsSearch.vue:374]: I'm not sure if this is new or not, but the labels also differ between `DetailsSearch` and `DetailsPanelEvent`
image

File: packages/components/src/components/FindingDetailsPanel.vue (hunk 3 above)
Comment (dustinbyrne) [reply packages/components/src/components/FindingDetailsPanel.vue:144]: The margins look like they could be more consistent here as well

File: packages/components/src/components/DetailsSearch.vue [hunk 4]
```diff
@@ -150,6 +158,29 @@ export default {
...
+        const { traceFilter } = fragment;
+
+        const events = fragment.traceFilter.split(' ').map((idStr) => {
+          const id = Number(idStr.split(':')[1]);
+          return eventsById[id];
+        });
+
+        items['analysis-finding'].data.push({
+          object: {
+            name: `${finding.impactDomain}:${finding.ruleTitle}`,
```
Comment (dustinbyrne) [inline packages/components/src/components/DetailsSearch.vue:176]: Missing a space (e.g. `Performance: N plus 1 SQL query`).

File: packages/components/src/components/DetailsSearch.vue [hunk 5]
```diff
@@ -150,6 +158,29 @@ export default {
...
+        return map;
+      }, {});
+
+      this.findings.forEach(({ finding, appMapUri: { fragment } }) => {
+        const { traceFilter } = fragment;
+
+        const events = fragment.traceFilter.split(' ').map((idStr) => {
+          const id = Number(idStr.split(':')[1]);
+          return eventsById[id];
+        });
```
Comment (dustinbyrne) [inline packages/components/src/components/DetailsSearch.vue:172]: IMO one of the major drivers for this feature is to replace the use of the trace filter for navigating events within a finding. We should drop this.

File: packages/components/src/components/DetailsSearch.vue [hunk 6]
```diff
@@ -264,57 +319,81 @@ export default {
...
     &-list {
       margin: 0;
-      padding: 0;
+      padding: 0 0.5rem;
       list-style: none;
+      li {
+        &:last-of-type {
+          border-bottom: 0;
+        }
+      }
```
[the remark this answers is not in the document]
Comment (dustinbyrne) [reply packages/components/src/components/DetailsSearch.vue]: It's a little more consistent now

[the remark this answers is not in the document]
Comment (dustinbyrne) [reply packages/components/src/components/DetailsSearch.vue]: ...maybe?

image
vs
image

File: packages/diagrams/src/scss/components/_diagram.scss [hunk 7]
```diff
@@ -108,7 +108,7 @@
         }
 
         &.database {
-          fill: $database;
+          fill: red;
```
[the remark this answers is not in the document]
Comment (typaulhus) [reply packages/diagrams/src/scss/components/_diagram.scss]: I pushed some fixes for the styling pieces you mentioned. :)

**Review by dustinbyrne — APPROVED** [review]

**Pull request merged**